internal/ota-api/dto.go

```diff
@@ -28,8 +29,9 @@ import (
 
 type (
 	OtaStatusResponse struct {
-		Ota    Ota     `json:"ota"`
-		States []State `json:"states,omitempty"`
+		FirmwareSize *int64  `json:"firmware_size,omitempty"`
```

Comment:
`omitempty` tags are only used during encoding, so I don't think you need them here.